std/container/rbtree.d

```diff
@@ -1094,6 +1094,110 @@ if (is(typeof(binaryFun!less(T.init, T.init))))
         assert(t1 != o);  // pathological case, must not crash
     }
 
+    /**
+     * Generates a hash for a Red Black Tree
+     *
+     *
+     */
+    override size_t toHash() nothrow @safe
+    {
+        import std.traits : hasMember;
+
+        auto thisRange = this[];
+
+        static if (hasMember!(Elem, "toHash"))
```

Comment:
What's the reason for having this branch instead of always using `typeid(x).getHash(&x)`?

Reply:
If the type of the elements in the rbtree implements `toHash` then this hashing function will be used in computing the hash of a node in the tree (I have written a unittest [here](https://github.com/dlang/phobos/pull/6517/files/63c9859cd2e024017a69a3d6aa82f1b33e7b8633#diff-5ce425da0dc54bc1f90d170bf3ae1d8fR1165) for this case).